### PR TITLE
Add inline select component

### LINKED ISSUE #152: Add Inline Select component
HTML and style: http://carbondesignsystem.com/components/select/code
Usage: http://carbondesignsystem.com/components/select/usage
Styling: http://carbondesignsystem.com/components/select/style

### PR DESCRIPTION
**Pull request by youda97**
closes #152

### CODE REVIEW

File: src/select/select.component.ts [hunk 1]
```diff
@@ -4,7 +4,9 @@ import { Component, Input } from "@angular/core";
 	selector: "ibm-select",
 	template: `
 		<div class="bx--form-item">
-			<div class="bx--select">
+			<div
+				[ngClass]="{ 'bx--select--inline': inlineSelect }"
```
Comment (zvonimirfras) [inline src/select/select.component.ts]: All other places in the codebase do
```suggestion
				[ngClass]="{'bx--select--inline': inlineSelect}"
```

File: src/select/select.component.ts [hunk 2]
```diff
@@ -18,6 +20,7 @@ import { Component, Input } from "@angular/core";
 })
 export class Select {
 	static selectCount = 0;
+	@Input() inlineSelect = false;
```
Comment (cal-smith) [inline src/select/select.component.ts]: I'd go with `inline = true` ... we know it's a select 😉

Comment (cal-smith) [reply src/select/select.component.ts]: Or even `type = "inline"` ... but I'm not so sure about that one

Comment (zvonimirfras) [reply src/select/select.component.ts]: Will we have other types of select?

Comment (cal-smith) [reply src/select/select.component.ts]: Probably not ... dropdown uses `type` for multi/single ... but even that is kinda weird (the actual `select` element uses a boolean `multiple` for multi/single)

File: src/select/select.stories.ts [hunk 3]
```diff
@@ -25,4 +25,20 @@ storiesOf("Select", module).addDecorator(
           </optgroup>
 		</ibm-select>
 	`
+	}))
+	.add("Inline", () => ({
+		template: `
+		<ibm-select inline="true">
```
Comment (panpan-lin) [inline src/select/select.stories.ts]: if you use `inline` here, the HTML Dom element for the Select component itself will have an extra attribute `inline` there. If you use `[inline]` instead, then the style will be applied at the right place without having the additional attribute added to the Dom note.

File: src/select/select.component.ts (hunk 2 above)
Comment (zvonimirfras) [reply src/select/select.component.ts]: @cal-smith Based on our Friday discussion, would this be
```suggestion
	@Input() display: "inline";
```
or something along those lines?

**Comment by cal-smith** [comment]
Also there's some merge conflicts ...

**Comment by youda97** [comment]
@cal-smith Why not just fo `@Input() display: "inline";`. Since its inline or its not. We never compare to `default`

**Comment by cal-smith** [comment]
Yeah we could do that ... Personally I prefer having an explicit default state (so there's no questions) but that works too

**Comment by zvonimirfras** [comment]
So are we agreeing that `"default"` is always an option if there's only one extra choice?

So here, we'd have `"default"` for explicit reasons, but for i.e. `"inline" | "single" | "multi"` we don't have a 4th `"default"` because (let's say) `"single"` is the default?

Do a 👍 or 👎

**Comment by cal-smith** [comment]
Yep I can get behind that, keeps things obvious

**Pull request merged**